website/content/docs/connect/gateways/api-gateway/configuration/api-gateway.mdx

```diff
@@ -0,0 +1,331 @@
+				{
+					Kind = "inline-certificate"
+					Name = "<name of inline-certificate>"
+					Namespace = "<enterprise: namespace of the certificate>"
+					Partition = "<enterprise: partition of the certificate>"
+				}
+			]
+		}
+	}
+]
+```
+
+```json
+{
+	"Kind": "api-gateway",
+	"Name": "<name of api gateway>",
+	"Namespace": "<enterprise: namespace of the gateway>",
+	"Partition": "<enterprise: partition of the gateway>",
+	"Meta": {
+		"<any key>": "<any value>"
+	},
+	"Listeners": [
+		{
+			"Name": "<unique name for this listener>",
+			"Port": <external service port>,
+			"Protocol": "<protocol used by external service>",
+			"TLS": {
+				"MaxVersion": "<version of TLS>",
+				"MinVersion": "<version of TLS>",
+				"CipherSuites": [
+					"<cipher suite>"
+				],
+				"Certificates": [
+					{
+						"Kind": "inline-certificate",
+						"Name": "<name of inline-certificate>",
+						"Namespace": "<enterprise: namespace of the certificate>",
+						"Partition": "<enterprise: partition of the certificate>"
+					}
+				]
+			}
+		}
+	]
+}
+```
+
+</CodeTabs>
+
+## Specification
+
+This section provides details about the fields you can configure in the
+`api-gateway` configuration entry.
+
+### `Kind`
+
+Specifies the type of configuration entry to implement. This must be
+`api-gateway`.
+
+#### Values
+
+- Default: none
+- This field is required.
+- Data type: string value that must be set to `"api-gateway"`.
+
+### `Name`
+
+Specifies a name for the configuration entry. The name is metadata that you can
+use to reference the configuration entry when performing Consul operations,
+such as applying a configuration entry to a specific cluster.
+
+#### Values
+
+- Default: none
+- This field is required.
+- Data type: string
+
+### `Namespace` <EnterpriseAlert inline />
+
+Specifies the Enterprise [namespace](/consul/docs/enterprise/namespaces) to apply to the configuration entry.
+
+#### Values
+
+- Default: `"default"` in Enterprise
+- Data type: string
+
+### `Partition` <EnterpriseAlert inline />
+
+Specifies the Enterprise [admin partition](/consul/docs/enterprise/admin-partitions) to apply to the config entry.
+
+#### Values
+
+- Default: `"default"` in Enterprise
+- Data type: string
+
+### `Meta`
+
+Specifies an arbitrary set of key-value pairs to associate with the gateway.
+
+#### Values
+
+- Default: none
+- Data type: map containing one or more keys and string values.
+
+### `Listeners[]`
+
+Specifies a list of listeners that gateway should set up. Listeners are
+uniquely identified by their port number.
+
+#### Values
+
+- Default: none
+- This field is required.
+- Data type: List of maps. Each member of the list contains the following fields:
+  - [`Name`](#listeners-name)
+  - [`Port`](#listeners-port)
+  - [`Hostname`](#listeners-hostname)
+  - [`Protocol`](#listeners-protocol)
+  - [`TLS`](#listeners-tls)
+
+### `Listeners[].Name`
+
+Specifies the unique name for the listener. This field accepts letters, numbers and hyphens.
```

Comment:
```suggestion
Specifies the unique name for the listener. This field accepts letters, numbers, and hyphens.
```

Reply:
Lol, you have to fight @boruszak because he suggested we remove this and you want us to add it back in.

Reply:
🥊 - I like the Oxford comma, but I won't die on this hill.